src/models/event-timeline-set.ts

```diff
@@ -119,13 +120,15 @@ export class EventTimelineSet extends TypedEventEmitter<EmittedEvents, EventTime
      * Set to true to enable improved timeline support.
      * @param {Object} [opts.filter = null]
      * The filter object, if any, for this timelineSet.
-     * @param {MatrixClient} client the Matrix client which owns this EventTimelineSet,
+     * @param {MatrixClient=} client the Matrix client which owns this EventTimelineSet,
```

Comment:
Is this only necessary for threads? That's the only place we use this functionality. We should add a commentthe reasoning for when this is necessary and the thread use case.

Reply:
Its used for the notification timeline (or any without a room) where a `client` is still necessary but cannot be accessed via `room.client`

Reply:
Can we add that statement to the JSDoc here? It's great context for why we use it.